src/hash_types.rs

```diff
@@ -20,59 +20,71 @@
 //! hash).
 //!
 
-use crate::hashes::{Hash, sha256, sha256d, hash160, hash_newtype};
-
 macro_rules! impl_hashencode {
     ($hashtype:ident) => {
         impl $crate::consensus::Encodable for $hashtype {
-            fn consensus_encode<S: $crate::io::Write>(&self, s: S) -> Result<usize, $crate::io::Error> {
+            fn consensus_encode<S: $crate::io::Write>(
+                &self,
+                s: S,
+            ) -> Result<usize, $crate::io::Error> {
```

Comment:
Having arguments split up is very hard to read for me. :( This line doesn't seem too long to me.

Reply:
Agreed the formatter does not play that nicely with macros, I added `#[rustfmt::skip]` to this one.

Reply:
I also shortened a bunch of function parameters `formatter: &mut Formatter` -> `f: &mut Formatter` to assist in keeping things on one line.

Reply:
Yeah, using `f` for formatter in `fmt::*` impls is pretty standard, so one-letter variable is accpetable.